ni_measurementlink_service/_internal/parameter/serialization_strategy.py

```diff
@@ -1,23 +1,34 @@
 """Serialization Strategy."""
 
-from typing import Any, Callable
+from typing import Any, Callable, Dict
 
 from google.protobuf import type_pb2
-from google.protobuf.internal import decoder
-from google.protobuf.internal import encoder
+from google.protobuf.descriptor import FieldDescriptor
+from google.protobuf.internal import decoder, encoder
+from typing_extensions import Buffer
 
+from ni_measurementlink_service._internal.parameter import _message
+from ni_measurementlink_service._internal.stubs.ni.protobuf.types import xydata_pb2
 
-def _scalar_encoder(encoder) -> Callable[[int], Callable]:
+LowLevelEncoder = Callable[[Callable[[Buffer], int], Any, None], None]
+EncoderFactory = Callable[[int], LowLevelEncoder]
+LowLevelDecoder = Callable[
+    [memoryview, int, int, type_pb2.Field.Kind.ValueType, Dict[FieldDescriptor, Any]], int
+]
+DecoderFactory = Callable[[int, str], LowLevelDecoder]
+
+
+def _scalar_encoder(encoder) -> EncoderFactory:
     """Abstract Specific Encoder(Callable) as Scalar Encoder Callable that takes in field index.
```

Comment:
I think the docstrings in this file do more harm than good. They are overly verbose, surface-level descriptions of the functions, and they are Littered with distracting Random Caps.

What does `_scalar_encoder` do? It performs partial application of is_repeated=False is_packed=False.

What does `_vector_encoder` do? It performs partial application of is_repeated=True and makes is_packed optional.

Do we need to list the arguments and their types in tedious Google docstring format? I'd rather have type hints in the function signature.

PEP8 says docstrings are not required for non-public methods. https://pep8.org/#documentation-strings

Reply:
Should be resolved by the refactoring PR.